MekHQ/src/mekhq/gui/dialog/IndividualHireDialog.java

```diff
@@ -0,0 +1,306 @@
+                refreshView();
+            }));
+        }
+
+        leftPanel.add(createButton("btnChoosePortrait", evt -> {
+            AbstractIconChooserDialog portraitDialog = new PortraitChooserDialog(getFrame(), getPerson().getPortrait());
+            if ((portraitDialog.showDialog() == JOptionPane.OK_OPTION) && (portraitDialog.getSelectedItem() != null)) {
+                getPerson().setPortrait(portraitDialog.getSelectedItem());
+                refreshView();
+            }
+        }));
+
+        if (getGUI().getCampaign().isGM()) {
+            leftPanel.add(createButton("btnEditPerson", evt -> {
+                final Gender gender = person.getGender();
+                CustomizePersonDialog customizePersonDialog = new CustomizePersonDialog(
+                        getFrame(), true, getPerson(), getGUI().getCampaign());
+                customizePersonDialog.setVisible(true);
+                if (gender != getPerson().getGender()) {
+                    getGUI().getCampaign().assignRandomPortraitFor(getPerson());
+                }
+                refreshView();
+            }));
+
+            leftPanel.add(createButton("btnRegenerate", evt -> createNewPerson()));
+        }
+
+        return leftPanel;
+    }
+
+    private Component createRightComponent() {
+        setScrollPerson(new JScrollPane(null, JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED,
+                JScrollPane.HORIZONTAL_SCROLLBAR_NEVER));
+        getScrollPerson().setName("scrollPerson");
+        getScrollPerson().setMinimumSize(new Dimension(450, 180));
+        getScrollPerson().setPreferredSize(new Dimension(450, 180));
+        return getScrollPerson();
+    }
+
+    @Override
+    protected JPanel createButtonPanel() {
+        JPanel panel = new JPanel(new GridLayout(1, getGUI().getCampaign().isGM() ? 3 : 2));
+        panel.add(createButton("Hire.text", "Hire.toolTipText", "hireButton", this::okButtonActionPerformed));
+        if (getGUI().getCampaign().isGM()) {
+            panel.add(createButton("AddGM.text", "AddGM.toolTipText", "addGMButton", evt -> {
+                if (getGUI().getCampaign().recruitPerson(getPerson(), true)) {
+                    createNewPerson();
+                }
+            }));
+        }
+        panel.add(createButton("Cancel.text", "Cancel.toolTipText", "cancelButton", this::cancelActionPerformed));
+        return panel;
+    }
+
+    @Override
+    protected void finalizeInitialization() {
+        super.finalizeInitialization();
+        createNewPerson();
+    }
+
+    @Override
+    protected void setCustomPreferences(final PreferencesNode preferences) {
+        super.setCustomPreferences(preferences);
+        preferences.manage(new JSplitPanePreference(getIndividualHireSplitPane()));
+        preferences.manage(new JComboBoxPreference(getComboPrimaryRole()));
+        updateComboRank(false); // this needs to occur post-primary role and pre-combo rank preferences

```

Comment:
this needs to occur between primary role and rank preferences to set it correctly

Reply:
I hate the location... rework this